sos/upload/targets/__init__.py

```diff
@@ -176,7 +269,12 @@ def pre_work(self, hook_commons):
                 self.prompt_for_upload_s3_endpoint()
                 self.prompt_for_upload_s3_access_key()
                 self.prompt_for_upload_s3_secret_key()
+            if self.get_proxy_host():
+                self.generate_proxy_list(use_prompts=True)
             self.ui_log.info('')
+        else:
+            if self.get_proxy_host():
+                self.generate_proxy_list(use_prompts=False)
```

Comment:
Code de-duplication: isnt it better to call (after the `if not cmdline_opts.batch ..` block):

```
if self.get_proxy_host():
    self.generate_proxy_list(use_prompts=(not cmdline_opts.batch and not cmdline_opts.quiet))
```

?

Well, then the `log_info` of empty line would be entered before the prompt, not welcomed, I guess?